src/lib.rs

```diff
@@ -244,6 +249,19 @@ impl SystemMode {
             .finish()
             .map_err(|e| Error::BufferTooSmall(Some(e)))
     }
+
+    async fn setup_psm(&self) -> Result<(), Error> {
+        if self.lte_support {
+            if self.lte_psm_support {
+                // Set Power Saving Mode (PSM)
+                at::send_at::<0>("AT+CPSMS=1").await?;
+            } else {
+                // Set Power Saving Mode (PSM)
+                at::send_at::<0>("AT+CPSMS=1").await?;
```

Comment:
You probably mean this here:

```suggestion
                // Turn off PSM
                at::send_at::<0>("AT+CPSMS=0").await?;
```